compiler/rustc_trait_selection/src/traits/util.rs

```diff
@@ -208,8 +209,22 @@ pub fn impl_subject_and_oblig<'a, 'tcx>(
     let predicates = predicates.instantiate(selcx.tcx(), impl_substs);
     let InferOk { value: predicates, obligations: normalization_obligations2 } =
         selcx.infcx.at(&ObligationCause::dummy(), param_env).normalize(predicates);
-    let impl_obligations =
-        super::predicates_for_generics(|_, _| ObligationCause::dummy(), param_env, predicates);
+    let cause_span = cause.map(|cause_def_id| selcx.tcx().def_span(cause_def_id));
+    let impl_obligations = super::predicates_for_generics(
+        move |_, span| {
+            if let Some(cause_def_id) = cause {
+                ObligationCause::new(
```

Comment:
this is more of a nit but I think using an `ImplDerivedObligation` instead of a `BindingObligation` tweaks the wording in that `'static` UI test a bit -- see #110874

Reply:
Lifetime error reporting does it's own handling of `ObligationCauses`, the current version is probably the best I can get without making changes there in this PR.